SpatialGDK/Source/SpatialGDKTests/SpatialGDK/Interop/SpatialWorkerFlags/SpatialWorkerFlagsTest.cpp

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+#include "Interop/SpatialWorkerFlags.h"
+#include "DummyObject.h"
+
+#include "TestDefinitions.h"
+
+
+#define SPATIALWORKERFLAGS_TEST(TestName) \
+	GDK_TEST(Core, SpatialWorkerFlags, TestName)
+
+namespace
+{
+	Worker_FlagUpdateOp Create_Worker_FlagUpdateOp(const char* name, const char* value)
+	{
+		Worker_FlagUpdateOp op = {};
+		op.name = name;
+		op.value = value;
+
+		return op;
+	}
+}
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_flagUpdate_op_WHEN_adding_a_worker_flag_THEN_flag_added)
+{
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+	FString OutValue;
+
+	//add test flag
+	Worker_FlagUpdateOp op = Create_Worker_FlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(op);
+
+	TestTrue("Flag added in the WorkerFlags map: ", SpatialWorkerFlags->GetWorkerFlag("test", OutValue));
+
+	return true;
+}
+
+
+SPATIALWORKERFLAGS_TEST(GIVEN_a_flagUpdate_op_WHEN_removing_a_worker_flag_THEN_flag_removed)
+{
+	USpatialWorkerFlags* SpatialWorkerFlags = NewObject<USpatialWorkerFlags>();
+	FString OutValue;
+
+	//add test flag
+	Worker_FlagUpdateOp op = Create_Worker_FlagUpdateOp("test", "10");
+	SpatialWorkerFlags->ApplyWorkerFlagUpdate(op);
+
```

Comment:
We should probably use a TestTrue for the flag here so that we know that it was actually added in the first place